docs/modules/ROOT/pages/api/fungibleToken.adoc

```diff
@@ -0,0 +1,314 @@
+==== `[.contract-item-name]#++_unsafeTransferFrom++#++(from: Either<ZswapCoinPublicKey, ContractAddress>, to: Either<ZswapCoinPublicKey, ContractAddress>, value: Uint<128>) → Boolean++` [.item-kind]#circuit#
+
+Unsafe variant of <<FungibleTokenModule-transferFrom,transferFrom>> which allows transfers to contract addresses.
+
+WARNING: Transfers to contract addresses are considered unsafe because contract-to-contract calls are not currently supported.
+Tokens sent to a contract address may become irretrievable.
+Once contract-to-contract calls are supported, this circuit may be deprecated.
+
+Requirements:
+
+- Contract is initialized.
+- `from` is not the zero address.
+- `from` must have a balance of at least `value`.
+- `to` is not the zero address.
+- The caller has an allowance of ``from``'s tokens of at least `value`.
+
+[.contract-item]
+[[FungibleTokenModule-_transfer]]
+==== `[.contract-item-name]#++_transfer++#++(from: Either<ZswapCoinPublicKey, ContractAddress>, to: Either<ZswapCoinPublicKey, ContractAddress>, value: Uint<128>) → []++` [.item-kind]#circuit#
+
+Moves a `value` amount of tokens from `from` to `to`.
+This circuit is equivalent to <<FungibleTokenModule-transfer,transfer>>, and can be used to e.g.
+implement automatic token fees, slashing mechanisms, etc.
+
+NOTE: Transfers to contract addresses are currently disallowed until contract-to-contract interactions are supported in Compact.
+This restriction prevents assets from being inadvertently locked in contracts that cannot currently handle token receipt.
+
+Requirements:
+
+- Contract is initialized.
+- `from` is not be the zero address.
+- `from` must have at least a balance of `value`.
+- `to` must not be the zero address.
+- `to` must not be a ContractAddress.
+
+[.contract-item]
+[[FungibleTokenModule-_unsafeUncheckedTransfer]]
+==== `[.contract-item-name]#++_unsafeUncheckedTransfer++#++(from: Either<ZswapCoinPublicKey, ContractAddress>, to: Either<ZswapCoinPublicKey, ContractAddress>, value: Uint<128>) → []++` [.item-kind]#circuit#
+
+Unsafe variant of <<FungibleTokenModule-_transfer,_transfer>> which allows transfers to contract addresses.
+
+WARNING: Transfers to contract addresses are considered unsafe because contract-to-contract calls are not currently supported. Tokens sent to a contract address may become irretrievable.
+Once contract-to-contract calls are supported, this circuit may be deprecated.
+
+Requirements:
+
+- Contract is initialized.
+- `from` is not the zero address.
+- `to` is not the zero address.
+
+[.contract-item]
+[[FungibleTokenModule-_update]]
+==== `[.contract-item-name]#++_update++#++(from: Either<ZswapCoinPublicKey, ContractAddress>, to: Either<ZswapCoinPublicKey, ContractAddress>, value: Uint<128>) → []++` [.item-kind]#circuit#
+
+Transfers a `value` amount of tokens from `from` to `to`,
+or alternatively mints (or burns) if `from` (or `to`) is the zero address.
+
+Requirements:
+
+- Contract is initialized.
```

Comment:
Non-blocking: This is the only circuit in the module that's not exported. We should explain why.

Reply:
Good call!